### PR TITLE
ROX-10014: Update preferred node affinity for central

### PR DESCRIPTION
**Pull request by kurlov**
## Description

[Ticket link](https://issues.redhat.com/browse/ROX-10014)

Central is recommended to run on infra or master nodes because Central is considered part of the OpenShift infrastructure.
Thus, Central `nodeAffinity` have to be switched to Infra->Compute->Master from Compute->Infra->Master

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

Please, let me know if this change worth a documentation upgrade. I think that it's not important enough.

## Testing Performed

Only unit tests

### CODE REVIEW

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ spec:
       affinity:
         nodeAffinity:
           preferredDuringSchedulingIgnoredDuringExecution:
-            # Central is single-homed, so avoid preemptible nodes.
+            # Central is single-homed, so avoid preemptive nodes.
```
Comment (porridge) [inline image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml:42]: Why this change? The key is `.../gke-preemptible`, not `...preemptive`.

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml [hunk 2]
```diff
@@ -48,18 +48,18 @@ spec:
                     values:
                     - "true"
             {{- if ._rox.env.openshift }}
-            - weight: 25
+            - weight: 50
```
Comment (porridge) [inline image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml:51]: Just curious why not keep it at 75?

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml [hunk 3]
```diff
@@ -48,18 +48,18 @@ spec:
...
               preference:
                 matchExpressions:
-                  - key: node-role.kubernetes.io/compute
+                  - key: node-role.kubernetes.io/infra
```
[the remark this answers is not in the document]
Comment (porridge) [reply image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml:54]: Does the order here matter? If not then I'm not sure it's worth moving it because the highest weight rule (100) is at the bottom.

**Review by porridge — APPROVED** [review]
A few nitpicks, but overall LGTM!
Thanks for adding the tests!

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml:54]: The order is by preferred note and not by weight, the master affinity has a `NotIn` operator

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml (hunk 1 above)
Comment (kurlov) [reply image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml]: Oops, I thought it was a typo

**Review by SimonBaeumer — APPROVED** [review]

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kurlov) [reply image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml:54]: Order in the yaml doesn't matter but I think it is nice that it reflects the priority.
Affinity logic was a bit confusion for me initially. Both `infra` and `compute` has `operator: In`. It means that we add its weight to the final score if such label is presented in the node. `Infra` has higher weight so that means that it is more preferable. `Master` has max weight but it also has `operator: NotIn` so it is sort of anti-affinity.  Basically it means that if we choose between master node and something else, we always choose non master node. That's why it is in the end of the list in the yaml

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml (hunk 2 above)
Comment (kurlov) [reply image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml:51]: It could be 75. Basically it could be any number higher than 25 (`compute`'s weight)
